source/compiler/qsc_project/src/project.rs

```diff
@@ -46,6 +47,10 @@ pub struct Project {
     pub errors: Vec<Error>,
     /// The type of project. This is used to determine how to load the project.
     pub project_type: ProjectType,
+    /// QIR target profile for this project (from manifest or default)
+    pub target_profile: Profile,
+    /// Whether this project is a single file (i.e. no manifest file).
+    pub is_single_file: bool,
```

Comment:
Just make `is_single_file` a field of `PackageGraphSources` . It logically and neatly belongs there. I think it'll save you a bunch of plumbing.

Also, personally I'd reverse the logic and call it `has_manifest` since I feel that that would be more accurate. (A project with a `qsharp.json` could still have a single file... so this name is a bit confusing IMO)